client-app/src/Components/NewItemForm.tsx

```diff
@@ -147,7 +201,11 @@ const NewItemForm: React.FC = () => {
             try {
                 const response = await axios.post(
                     `${process.env.REACT_APP_BACKEND_API_BASE_URL}donatedItem`,
-                    formData,
+                    {
+                        ...formData,
+                        donorId: formData.donorId,
+                        programId: formData.programId,
+                    },
```

Comment:
Look at the payload sent for **POST** `/donatedItem` request. It has null values for `donorId` and `programId`, this should always be integer (which should be the primary key of selected donor or program) .
Please do not send program name and donor email, remove all such variables from this file

![image](https://github.com/user-attachments/assets/cd2c65cc-424e-433e-bcf1-05bccc8106c0)